dodder-common/src/test/java/cc/dodder/common/entity/TreeTest.java

```diff
@@ -0,0 +1,107 @@
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertFalse;
+
+public class TreeTest {
+
+  @Test
+  public void testCheckExist() {
+    final Tree objectUnderTest = new Tree();
+    final Node node = new Node();
+    node.addChild(new Node());
+    assertFalse(objectUnderTest.checkExist(node, null));
+  }
+
+  @Test
+  public void testConstructor() {
+    final Tree tree = new Tree("foo");
+    assertEquals(101605, tree.getRoot().hashCode());
+    assertEquals(-1, tree.getRoot().getIndex());
+    assertEquals(-1, tree.getRoot().getPid());
+    assertEquals(0, tree.getRoot().getNid());
+    assertEquals("foo", tree.getRoot().getFilename());
+    assertNull(((Node)tree.getRoot()).getChildren());
+    assertNull(tree.getRoot().getFilesize());
+  }
+
+  @Test
+  public void testCreateTree() {
+    final Tree tree = new Tree();
+    final ArrayList<Node> nodes = new ArrayList<>();
+    tree.setRoot(new Node());
+    nodes.add(new Node(0, 0, "foo", 0l, 0));
+    nodes.add(new Node(1, 1, "bar", 0l, 1));
+    tree.createTree(nodes);
+    assertEquals(new Node(0, 0, "foo", 0l, 0), tree.getRoot().getChildren().get(0));
+  }
+
+  @Test
+  public void testGetHtmlNoChildren() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "a\'b\'c", 0l, 0));
+    assertEquals(
+            "<li><span class=\"file\">a'b'c<small>(0 B)</small></span></li>",
+            tree.getHtml(new Node(0, 0, "a\'b\'c", 0l, 0)));
+  }
+
+  @Test
+  public void testGetHtmlMatchingPid() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "a\'b\'c", 0l, 0));
+    final Node node = new Node(0, 0, "a/b/c", 0l, 0);
+    node.setChildren(new ArrayList<Node>());
+    assertEquals(
+            "<ul class=\"filetree treeview\"><p><span class=\"bticon\">a\'b\'c</span></p></ul></li>",
+            tree.getHtml(node));
+  }
+
+  @Test
+  public void testGetHtmlNotMatchingPid() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "a\'b\'c", 0l, 0));
+    final Node node = new Node(-2_147_483_648, 0, "a/b/c", 0l, 0);
+    node.setChildren(new ArrayList<Node>());
+    assertEquals("<li class=\"closed\"><span class=\"folder\">a/b/c</span><ul></ul></li>",
+            tree.getHtml(node));
+  }
+
+  @Test
+  public void testGetLeafList() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "foo", 0l, 0));
+    final List<Node> nodes = tree.getLeafList();
+    assertEquals(1, nodes.size());
+    assertEquals(0, nodes.get(0).getIndex());
+    assertEquals(0, nodes.get(0).getPid());
+    assertEquals(0, nodes.get(0).getNid());
+    assertEquals("foo", nodes.get(0).getFilename());
+  }
+
+  @Test
+  public void testGetLeafListChildrenNotNull() {
+    final Tree tree = new Tree();
+    final Node node = new Node();
+    node.setChildren(new ArrayList());
+    tree.setRoot(node);
+    assertEquals(new ArrayList<Node>(), tree.getLeafList());
```

Comment:
There's an `assertArrayEquals` method which might work better here? I think it might work with lists as well.